interface/reports/immunization_report.php

```diff
@@ -494,6 +512,27 @@ class='datepicker form-control'
             </div>
         <?php } ?>
     </form>
-
+    <script>
+        
+        function exportData() {
+            let data = <?php echo js_escape(json_encode($rows ?? '')); ?>;
```

Comment:
note js_escape() is a wrapper for json_encode(), so only need to do one here.
I generally use  json_encode() when operating on a php array as here, so would just drop the js_escape call and keep the json_encode() call.

Other than this, code looks ready :)